client/post-editor/editor-sidebar/util.js

```diff
@@ -0,0 +1,12 @@
+/**
+ * External dependencies
+ */
+import { PropTypes } from 'react';
+
+export const NESTED_SIDEBAR_NONE = 'NESTED_SIDEBAR_NONE';
```

Comment:
The idea of a nested sidebar is to replace the main content of the sidebar, but keep some consistency across the header (slightly updated to get some navigation behaviors) and the footer.

Replacing the main content to display more specific information (or advanced options) isn't tied to revisions, it's a more generic behavior not related to "revisions" per se, hence why I try to keep most of the naming independent from the current use case of displaying a list of revisions.

The current implementation is a simple state machine driven by the `nestedSidebar` value in the `PostEditor`'s state (lowest common denominator for all the components that need visual updates depending on the state of the sidebar: `EditorGroundControl`, `EditorActionBar`, `EditorSidebar`). In the default state `NESTED_SIDEBAR_NONE`, we get the parent `SidebarRegion` (which contains the current sidebar, with its drawers), in other states (like `NESTED_SIDEBAR_REVISIONS`), the parent `SidebarRegion` is hidden and the nested `SidebarRegion` takes its place.

Though we could (slightly) simplify the cognitive load by renaming most `nested` usage with `revisions`, and using a boolean instead of a string-enum in the state `nestedSidebar` (renamed `revisionSidebar`), it would tightly couple the idea of a nested sidebar and its current usage for revisions. The current implementation has the advantage of being more future-proof, which is a very debatable property tbh :) but considering the added cognitive load is quite small, the decoupling is worth it imho.